features/device_key/source/DeviceKey.h

```diff
@@ -139,16 +139,16 @@ class DeviceKey : private mbed::NonCopyable<DeviceKey> {
     int get_derived_key(uint32_t *ikey_buff, size_t ikey_size, const unsigned char *isalt, size_t isalt_size,
                         unsigned char *output, uint32_t ikey_type);
 
-    /** Generate a random ROT key by using entropy
-     * @param output Output buffer for the generated key.
-     * @param size Input: The size of the buffer. If size is less
-     *                    than 16 bytes, the method generates an
-     *                    error. 16-31 bytes creates a 16-byte key.
-     *                    32 or higher generates a 32-byte key
-     *             Output: The actual written size to the buffer
-     * @return 0 on success, negative error code on failure
+    /** Generate Root of Trust from random sources.
```

Comment:
~from random sources~ the sources are not really random and you explain it below anyway.

Reply:
Corrected.
The reason is  IOTSTOR-941 Jira ticket